pkg/controller/common/manage_children.go

```diff
@@ -113,13 +113,13 @@ func revertField(newObj, orig *unstructured.Unstructured, fieldPath ...string) e
 	return nil
 }
 
-func MakeControllerRef(parent *unstructured.Unstructured) *metav1.OwnerReference {
+func MakeControllerRef(parent *unstructured.Unstructured, controller bool) *metav1.OwnerReference {
 	return &metav1.OwnerReference{
 		APIVersion:         parent.GetAPIVersion(),
 		Kind:               parent.GetKind(),
 		Name:               parent.GetName(),
 		UID:                parent.GetUID(),
-		Controller:         pointer.Bool(true),
+		Controller:         pointer.Bool(controller),
```

Comment:
I am still unsure if this is correct, looking at https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/controller/controllerutil/controllerutil.go#L59-L120 it seems that they set `blockOwnerDeletion` along with `controller`.

Reply:
reading https://github.com/kubernetes/design-proposals-archive/blob/main/api-machinery/controller-ref.md this field was introduced to avoid fighting over resources, so metacontroller would be creating object which it do not suppose to manage further